### ISSUE TITLE
MIRI s3d (and NIRSpec IFU) has OBSGEO but not MJD-OBS in FITS WCS, causing specutils.Spectrum1D to crash

### ISSUE THREAD

**Issue by pllim** [spacetelescope/jwst#6504]
I came to this conclusion based on the following files you can obtain from @havok2063 .

* Good WCS: `test_data/0.17.1/jw00736-o039_t001_miri_ch1-long_s3d.fits` (from `jwst` 0.17.1)
* Bad WCS: `test_data/1.3.2/jw00619-o094_t001_miri_ch1-long_s3d.fits` (from `jwst` 1.3.2)

```python
>>> import numpy as np
>>> from astropy import units as u
>>> from astropy.io import fits
>>> from astropy.wcs import WCS
>>> from specutils import Spectrum1D
>>> filename_good = 'jw00736-o039_t001_miri_ch1-long_s3d.fits'
>>> flux = np.ones(20).reshape((2, 2, 5)) * u.mJy  # The data itself does not matter
>>> pf_good = fits.open(filename_good)
>>> w_good = WCS(pf_good[1].header, pf_good)
WARNING: FITSFixedWarning: 'datfix' made the change 'Set DATE-BEG to '2020-04-11T22:42:07.487' from MJD-BEG.
Set DATE-AVG to '2020-04-11T23:13:10.030' from MJD-AVG.
Set DATE-END to '2020-04-11T23:44:29.033' from MJD-END'. [astropy.wcs.wcs]
>>> w_good.wcs.mjdobs
nan
>>> w_good.wcs.obsgeo
array([nan, nan, nan, nan, nan, nan])
>>> spec_good = Spectrum1D(flux=flux, wcs=w_good)
Input WCS indicates that the spectral axis is not last. Reshaping arrays to put spectral axis last.
>>> filename_bad = 'jw00619-o094_t001_miri_ch1-long_s3d.fits'
>>> pf_bad = fits.open(filename_bad)
>>> w_bad = WCS(pf_bad[1].header, pf_bad)
WARNING: FITSFixedWarning: 'datfix' made the change 'Set DATE-BEG to '2020-07-15T09:52:14.994' from MJD-BEG.
Set DATE-AVG to '2020-07-15T09:57:06.997' from MJD-AVG.
Set DATE-END to '2020-07-15T10:01:59.020' from MJD-END'. [astropy.wcs.wcs]
WARNING: FITSFixedWarning: 'obsfix' made the change 'Set OBSGEO-L to   -80.369340 from OBSGEO-[XYZ].
Set OBSGEO-B to   -14.072121 from OBSGEO-[XYZ].
Set OBSGEO-H to 1275517195.357 from OBSGEO-[XYZ]'. [astropy.wcs.wcs]
>>> w_bad.wcs.mjdobs
nan
>>>  w_bad.wcs.obsgeo
array([ 2.08020863e+08, -1.22590339e+09, -3.11673871e+08, -8.03693403e+01,
       -1.40721206e+01,  1.27551720e+09])
>>> spec_bad = Spectrum1D(flux=flux, wcs=w_bad)
Input WCS indicates that the spectral axis is not last. Reshaping arrays to put spectral axis last.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
.../astropy/time/core.py in _get_time_fmt(self, val, val2, format, scale, precision, in_subfmt, out_subfmt)
    441             try:
--> 442                 return cls(val, val2, scale, precision, in_subfmt, out_subfmt)
    443             except UnitConversionError:

.../astropy/time/formats.py in __init__(self, val1, val2, scale, precision, in_subfmt, out_subfmt, from_jd)
    118         else:
--> 119             val1, val2 = self._check_val_type(val1, val2)
    120             self.set_jds(val1, val2)

.../astropy/time/formats.py in _check_val_type(self, val1, val2)
    415         if val1.dtype.kind == 'f':
--> 416             val1, val2 = super()._check_val_type(val1, val2)
    417         elif (not orig_val2_is_none

.../astropy/time/formats.py in _check_val_type(self, val1, val2)
    253         if not (ok1 and ok2):
--> 254             raise TypeError('Input values for {} class must be finite doubles'
    255                             .format(self.name))

TypeError: Input values for mjd class must be finite doubles

The above exception was the direct cause of the following exception:

.../specutils/spectra/spectrum1d.py in __init__(self, flux, spectral_axis, wcs, velocity_convention, rest_value, redshift, radial_velocity, bin_specification, **kwargs)
    284                     spec_axis = self.wcs.pixel_to_world(np.arange(self.flux.shape[-1]))
    285                 else:
--> 286                     spec_axis = self.wcs.spectral.pixel_to_world(np.arange(self.flux.shape[-1]))
    287             else:
    288                 spec_axis = self.wcs.pixel_to_world(np.arange(self.flux.shape[-1]))

.../astropy/wcs/wcsapi/high_level_api.py in pixel_to_world(self, *pixel_arrays)
    322             world_values = (world_values,)
    323
--> 324         pixel_values = values_to_high_level_objects(*world_values, low_level_wcs=self.low_level_wcs)
    325
    326         if len(pixel_values) == 1:

.../astropy/wcs/wcsapi/high_level_api.py in values_to_high_level_objects(low_level_wcs, *world_values)
    258     """
    259     # Cache the classes and components since this may be expensive
--> 260     components = low_level_wcs.world_axis_object_components
    261     classes = low_level_wcs.world_axis_object_classes
    262

.../astropy/wcs/wcsapi/fitswcs.py in world_axis_object_components(self)
    329     @property
    330     def world_axis_object_components(self):
--> 331         return self._get_components_and_classes()[0]
    332
    333     @property

.../astropy/wcs/wcsapi/fitswcs.py in _get_components_and_classes(self)
    421
    422                 earth_location = EarthLocation(*self.wcs.obsgeo[:3], unit=u.m)
--> 423                 obstime = Time(self.wcs.mjdobs, format='mjd', scale='utc',
    424                                location=earth_location)
    425                 observer_location = SkyCoord(earth_location.get_itrs(obstime=obstime))

.../astropy/time/core.py in __init__(self, val, val2, format, scale, precision, in_subfmt, out_subfmt, location, copy)
   1529                 self._set_scale(scale)
   1530         else:
-> 1531             self._init_from_vals(val, val2, format, scale, copy,
   1532                                  precision, in_subfmt, out_subfmt)
   1533             self.SCALES = TIME_TYPES[self.scale]

.../astropy/time/core.py in _init_from_vals(self, val, val2, format, scale, copy, precision, in_subfmt, out_subfmt)
    382
    383         # Parse / convert input values into internal jd1, jd2 based on format
--> 384         self._time = self._get_time_fmt(val, val2, format, scale,
    385                                         precision, in_subfmt, out_subfmt)
    386         self._format = self._time.name

.../astropy/time/core.py in _get_time_fmt(self, val, val2, format, scale, precision, in_subfmt, out_subfmt)
    448                 # easier for user to see what is wrong.
    449                 if len(formats) == 1:
--> 450                     raise ValueError(
    451                         f'Input values did not match the format class {format}:'
    452                         + os.linesep

ValueError: Input values did not match the format class mjd:
TypeError: Input values for mjd class must be finite doubles
```

https://github.com/astropy/astropy/blob/2158c36979a5c8e4890a64947cfc631d1e928a72/astropy/wcs/wcsapi/fitswcs.py#L418-L424

Version info:

* Python 3.9.7
* `numpy` 1.21.2
* `astropy` 5.1.dev249+g2158c3697
* `specutils` 1.5.1.dev14+g12ff269c

xref spacetelescope/jdaviz#690

cc @larrybradley @nden

**Comment by hbushouse** [comment]
I'm no expert in the FITS time conventions, but a quick look at the defining paper (https://www.aanda.org/articles/aa/pdf/2015/02/aa24653-14.pdf) shows that keywords DATE-OBS and MJD-OBS are *allowed* in FITS headers, but doesn't explicitly say that they're required (at least I didn't find anything indicating that). A while back the keywords MJD-BEG, MJD-AVG, and MJD-END were introduced (see JWSTKD-88, JWSTKD-89, JWSTKD-90), in order to get JWST FITS headers onto the time convention standards, but no request was made for MJD-OBS. I'm guessing that was perhaps due to the fact that both DATE-OBS and MJD-OBS have always been a bit ambiguous as to their meaning, in terms of what part of the observation they refer to (beginning, mid-point, ...). The MJD-[BEG|AVG|END] keywords are obviously more explicit in that regard.

So, bottom line, I'm wondering if MJD-OBS is really required or is the specutils loader being overly picky?

**Comment by pllim** [comment]
The error is actually inside `astropy.wcs` but `specutils` triggered it. Why does your data have OBSGEO but not MJD-OBS?

**Comment by pllim** [comment]
@havok2063 also reported same problem with NIRSpec IFU cubes.

**Comment by nden** [comment]
I've looked into this and it's an astropy issue. I'm talking to someone to understand if there's a workaround.

**Comment by hbushouse** [comment]
The "OBSGEO-[X|Y|Z]" and "MJD-[BEG|AVG|END]" keywords are there due to requests (a couple years ago) from @RAShaw-sci (see, e.g. https://jira.stsci.edu/browse/JWSTKD-88, https://jira.stsci.edu/browse/JWSTKD-89, https://jira.stsci.edu/browse/JWSTKD-90, https://jira.stsci.edu/browse/JWSTKD-94, https://jira.stsci.edu/browse/JWSTKD-95, https://jira.stsci.edu/browse/JWSTKD-96). MJD-OBS was never requested and as far as I can tell is not required.

**Comment by havok2063** [comment]
@nden Do you know what kind of astropy issue this is? Or if this is a bug, when was it introduced?  I first started noticing this error back in June 2021, which looks between astropy versions `4.2.1` and `4.3`.

**Comment by pllim** [comment]
This problem exists as far back as astropy 4.1. I think things broke when OBSGEO is added to the header without MJD-OBS. 💭

**Comment by nden** [comment]
If I understand WCS Paper III correctly  `OBSGEO` keywords are used with `MJD-AVG` or `DATE-AVG` to define a location in the Terrestrial system, where `MJD-AVG` is the effective time of the observation. The `EarthLocation` object in the WCS API functions uses `MJD-OBS` which is what causes the error. [Here's a PR](https://github.com/astropy/astropy/pull/12598) which fixes this in astropy.

**Comment by pllim** [comment]
Thanks! I guess we'll have to do a local workaround until astropy/astropy#12598 is folded into the stack.
